rule/rule.go

```diff
@@ -369,6 +369,17 @@ func (f *File) Format() []byte {
 	return bzl.Format(f.File)
 }
 
+// Sort rules in the macro of this File. It doesn't sort the rules if
+// this File does not have a macro, e.g., WORKSPACE.
+// This method calls Sync internally.
+func (f *File) Sort() {
+	f.Sync()
+	//TODO: sort load statements too
+	if f.function != nil {
```

Comment:
I still think this condition should be removed. `update-repos.go` can check `f.DefName != ""` before deciding whether to call `Sort`.

Reply:
This method needs to use `f.function.stmt.Body`. If this condition is removed, what should it do if it is called with `function == nil`? Or should we assume the caller knows the precondition of this method?

Reply:
I was going to suggest it should be able to sort targets in a regular build files... but that's probably more complicated than necessary, since it would need to handle load statements and other expressions.

I think my objection is that if I'm reading code that calls `Sort`, I'd really expect it to be able to sort targets in any file. The name gives no hint it's limited to macros.

Perhaps instead, this should be renamed to `SortMacro` to make the limitation more obvious, and it should panic if it's called on something that's not a macro file.

Reply:
yeah, let's limit the scope of this PR to macros. It's less clear whether we should sort other files